packages/pyright-internal/src/common/workersHost.ts

```diff
@@ -0,0 +1,83 @@
+// This could be a more general abstractiopn of the host environment but for
+// now it's focussed on abstracting worker_threads vs Web Workers.
+//
+// For minimal distruption to the existing NodeJS code we use Node-type
+// events rather than MessageEvent.
+
+export type Transferable = ArrayBuffer | MessagePort;
+
+export interface MessageSourceSink {
+    // As per worker_thread, except any transferables may only be
+    // `value` itself or one level nested in an object or array value.
+    postMessage(value: any, transferList?: Transferable[]): void;
+    on(type: 'message' | 'error' | 'exit', listener: (data: any) => void): void;
+}
+
+export interface MessagePort extends MessageSourceSink {
+    start(): void;
+    close(): void;
+}
+
+export interface MessageChannel {
+    port1: MessagePort;
+    port2: MessagePort;
+}
+
+export interface WorkersHost {
+    parentPort(): MessagePort | null;
+    createWorker(initialData?: any): MessageSourceSink;
+    createMessageChannel(): MessageChannel;
+    threadId(): string;
+}
+
+// For now we default to the NodeJS one.
+// Need to review the impact of removing this on tests.
```

Comment:
Remove comment - I did remove the defaulting and at least the internal project's tests are fine.